### DISCUSSION TITLE
Private channel on telegram service.

### DISCUSSION THREAD

**Discussion by JuezFenix** [containrrr/shoutrrr#128] [Q&A] [answered]
Hi, its this posible?

If not, please should be a good feature.

Regards

**Comment by piksel** [comment, marked as answer]
Yes! To send to the channel "foo" you can use:
`telegram://TOKEN@telegram?channels=@foo`

The channel "username" (as telegram calls it) can be seen by selecting "View channel info":
![image](https://user-images.githubusercontent.com/807383/104808338-d05fd200-57e5-11eb-8423-726b89edca24.png)
It's the part after `t.me/` (shoutrrr in our channels case).

For other types of chats, it's a bit more complicated, since you need the chat id to send to it. I am currently working on a generator that will solve that problem...

**Reply by hugalafutro** [reply]
Hi, is there any news on the sending to other types of chat like bot? Ie with the token + chat id? It's the only way I know, I create bot, and use my telegram user chat id + the bot's token in apps such as radarr sonarr etc.

I have created a new telegram bot for container-mon, but can't figure out what a channel is in my case. I tried
`telegram://TOKEN@telegram?channels=@BOT_NAME` and `telegram://TOKEN@telegram?channels=BOT_NAME` neither of which work.

I don't get the channel, isn't it just group chat? Can I alternatively create a channel with only me in it to make this work?

Thanks for any suggestion!

**Reply by piksel** [reply]
Channels are separate from group chats and is more like a broadcast than a chat. I.E. only a few can actually post there, the rest are just "observers" if that makes sense. 
In the desktop client it's just beneath "New Group" in the main menu:
![image](https://user-images.githubusercontent.com/807383/115967160-48e6d500-a531-11eb-879b-413afe3478ae.png)

If you actually know the Chat ID, then you can just supply that instead of the @channel, like so:
```
telegram://TOKEN@telegram?channels=27358239252579,72374922623629
```

The only problem is that there is no good way to find out what those Chat ID's are from the client AFAIK.

**Reply by hugalafutro** [reply]
Hey, many thanks it worked! the chat id is easy-ish to find in web browser, you have to say something to the bot, then you go to `https://api.telegram.org/botAPPEND TOKEN HERE/getUpdates` and it spits out what you said to the bot in json and next to your name is the chat id.

**Reply by piksel** [reply]
Hah, yeah. I made an attempt to do this via a chat bot, and it worked okay-ish, but that is actually much simpler.

It could basically just be:
```
$ shoutrrr generate telegram
Token: ***********
Recent incoming chats for bot:
1) @user1: Hello bot! (Chat ID: 9208980325)
2) @user2: Hello bot! (Chat ID: 2935809205)

Select chats to be included [1,2]: 1,2

Done! Use the following URL:
telegram://***********@telegram?channels=9208980325,2935809205
```

**Reply by wegotourselvesareader** [reply]
This evening I was experimenting with getting watchtower to send me notfications via shoutrrr, and I found this page to be very helpful in finding the channel ID for a Telegram private channel: https://stackoverflow.com/questions/33858927/how-to-obtain-the-chat-id-of-a-private-telegram-channel

For me, the easiest way was to follow the steps on that page for using https://web.telegram.org, extracting the channel ID from the browser's URL bar, then prefixing the channel ID with "-100". The (redacted) result for supplying to shoutrrr is:

telegram://`<TOKEN>`@telegram/?channels=`-100xxxxxxxxxx`
